edx/edx-enterprise
# [ENT-290] Enable switch in Enterprise setup to hide Audit track

**Description:** This PR adds a boolean flag "Hide audit track" to the EnterpriseCustomer model and adjusts the new landing page to filter course modes accodingly.

**JIRA:** [ENT-290](https://openedx.atlassian.net/browse/ENT-290)

**Dependencies:**  
[PR on edx/edx-platform for the new `ENTERPRISE_COURSE_ENROLLMENT_AUDIT_MODES` setting](https://github.com/edx/edx-platform/pull/15095)

** Screenshots:**
![1](https://cloud.githubusercontent.com/assets/560781/25966911/aadbae12-3649-11e7-9eef-12d1917ca690.png)
![2](https://cloud.githubusercontent.com/assets/560781/25966910/aadb5f0c-3649-11e7-8cbf-2ef3835eb37c.png)
![3](https://cloud.githubusercontent.com/assets/560781/25966909/aad9f3d8-3649-11e7-94f0-b762a56bda0f.png)
![4](https://cloud.githubusercontent.com/assets/560781/25966912/aadf4a5e-3649-11e7-979e-ad13631ca44c.png)

**Testing instructions:**

1. Add an Enterprise Customer in the Django admin. Check its ID
2. Add a course in the customer catalog. Check its ID.
3. Visit the new landing page for the enterprise customer and course by typing a URL like this:

http://localhost:8000/enterprise/7c3c286a-9834-46ae-809e-e01b114ba980/course/course-v1:edX+DemoX+Demo_Course/enroll/

where 7c3c286a-9834-46ae-809e-e01b114ba980 should be the ID of the customer and 'course-v1:edX+DemoX+Demo_Course' should be the ID of the course

4. Check that the landing page shows both course modes (Enroll and Audit).
5. Go to the Enterprise Customer and check the "Hide audit track" flag
6. Go back to the landing page and verify that the Audit track is now hidden

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.~

**Post merge:**

**Reviewers**
